### PR TITLE
fixing pep8 E275 in rings/ and combinat/

### PR DESCRIPTION
**Pull request by fchapoton**
This is fixing some pycodestyle warnings about E275 in rings/ and combinat/ folders

(note that these warning are only seen with recent version of pycodestyle)

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/combinat/binary_tree.py [hunk 1]
```diff
@@ -914,7 +914,7 @@ def make_node(self, child_list=[None, None]):
         """
         self._require_mutable()
         child_lst = [self.__class__(self.parent(), x) for x in child_list]
-        if not(len(child_lst) == 2):
+        if not (len(child_lst) == 2):
```
Comment (dcoudert) [inline src/sage/combinat/binary_tree.py]: `if len(child_lst) != 2:` ?

File: src/sage/combinat/binary_tree.py [hunk 2]
```diff
@@ -1873,7 +1873,7 @@ def left_border_symmetry(self):
         labelled = self in LabelledBinaryTrees()
         labels = []
         t = self
-        while(t):
+        while (t):
```
Comment (dcoudert) [inline src/sage/combinat/binary_tree.py]: `while t:`

File: src/sage/combinat/cluster_algebra_quiver/cluster_seed.py [hunk 3]
```diff
@@ -4345,7 +4345,7 @@ def oriented_exchange_graph(self):
                     j = i.mutate(k, inplace=False)
                     Varj = tuple(sorted(j.cluster()))
                     covers.append((Vari, Varj))
-                    if not(Varj in known_clusters):
+                    if not (Varj in known_clusters):
```
Comment (dcoudert) [inline src/sage/combinat/cluster_algebra_quiver/cluster_seed.py]: `Varj not in known_clusters`

File: src/sage/combinat/cluster_algebra_quiver/quiver.py [hunk 4]
```diff
@@ -2180,7 +2180,7 @@ def d_vector_fan(self):
         from sage.geometry.fan import Fan
         from sage.geometry.cone import Cone
 
-        if not(self.is_finite()):
+        if not (self.is_finite()):
```
Comment (dcoudert) [inline src/sage/combinat/cluster_algebra_quiver/quiver.py:2183]: `not self.is_finite()` without extra parenthesis

File: src/sage/combinat/derangements.py [hunk 5]
```diff
@@ -445,10 +445,10 @@ def _rand_der(self):
         mark = [x < 0 for x in A]
         i, u = n, n
         while u >= 2:
-            if not(mark[i - 1]):
+            if not (mark[i - 1]):
```
Comment (dcoudert) [inline src/sage/combinat/derangements.py]: `not mark[i - 1]`

File: src/sage/combinat/tamari_lattices.py [hunk 6]
```diff
@@ -215,9 +215,9 @@ def GeneralizedTamariLattice(a, b, m=1, check=True):
     .. [BMFPR] \M. Bousquet-Melou, E. Fusy, L.-F. Preville Ratelle.
        *The number of intervals in the m-Tamari lattices*. :arxiv:`1106.1498`
     """
-    if not(gcd(a, b) == 1 and a >= b):
+    if not (gcd(a, b) == 1 and a >= b):
         raise ValueError("the numbers a and b must be coprime with a>=b")
-    if not(a >= b * m):
+    if not (a >= b * m):
```
Comment (dcoudert) [inline src/sage/combinat/tamari_lattices.py]: `if a < b * m:` ?

File: src/sage/combinat/words/abstract_word.py [hunk 7]
```diff
@@ -641,7 +641,7 @@ def _to_integer_iterator(self, use_parent_alphabet=False):
             mapping = {}
             next_value = 0
             for letter in self:
-                if not(letter in mapping):
+                if not (letter in mapping):
```
Comment (dcoudert) [inline src/sage/combinat/words/abstract_word.py]: `letter not in mapping`

File: src/sage/combinat/words/word_infinite_datatypes.py [hunk 8]
```diff
@@ -237,8 +237,8 @@ def __getitem__(self, key):
         if isinstance(key, slice):
             # Infinite words
             if self._len is Infinity or self._len is None:
-                if not(key.start is None) and key.start < 0 or \
-                        not(key.stop is None) and key.stop < 0:
+                if not (key.start is None) and key.start < 0 or \
```
Comment (dcoudert) [inline src/sage/combinat/words/word_infinite_datatypes.py]: `key.start is not None`

**Review by dcoudert — COMMENTED** [review]
In many places, you changed `not(foo)` to `not (foo)`. The extra parenthesis could be removed. Same with `while(foo):`, `return(foo)`, etc.

You could change that here to avoid a second dedicated PR.

**Comment by fchapoton** [comment]
J'en ai corrigé un certain nombre, mais c'est assez herculeen de le faire à la main..

**Comment by dcoudert** [comment]
> J'en ai corrigé un certain nombre, mais c'est assez herculeen de le faire à la main..
oui, je me doute. Ma remarque était surtout pour les lignes que tu as modifié, pas sur tout le code.

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by dcoudert** [comment]
More can be done, but not all in the same PR.

**Pull request merged**